Lib/Emscripten/Emscripten.cpp

```diff
@@ -166,22 +166,46 @@ DEFINE_INTRINSIC_FUNCTION(env, "_pthread_cond_broadcast", I32, _pthread_cond_bro
 {
 	return 0;
 }
-DEFINE_INTRINSIC_FUNCTION(env, "_pthread_key_create", I32, _pthread_key_create, I32 a, I32 b)
+
+static HashMap<I32, I32> pthreadSpecific = {};
+static int pthreadSpecificNextKey = 0;
+
+DEFINE_INTRINSIC_FUNCTION_WITH_MEM_AND_TABLE(env, "_pthread_key_create", I32, _pthread_key_create, I32 key, I32 destructorPtr)
 {
-	throwException(Runtime::Exception::calledUnimplementedIntrinsicType);
+	if (key == 0) {
+		// ErrNo codes - einval
+		return 22;
+	}
+
+	MemoryInstance* memory
+		= Runtime::getMemoryFromRuntimeData(contextRuntimeData, defaultMemoryId.id);
+	memoryRef<U32>(memory, key >> 2) = pthreadSpecificNextKey;
```

Comment:
Note: I'm not sure if this line is equivalent to the emscripten `HEAP32[key >> 2] = PTHREAD_SPECIFIC_NEXT_KEY;`... I would love to get feedback here!

Reply:
`HEAP32[key >> 2]` should translate to just `memoryRef<U32>(memory, key)`. `HEAP32[i]` implicitly multiples `i` by 4 when computing the memory address, so the `>> 2` is just undoing that.

Reply:
Got it! Thanks for the explanation :)